clang/lib/CConv/CheckedRegions.cpp

```diff
@@ -168,105 +240,130 @@ bool CheckedRegionFinder::isInStatementPosition(CallExpr *C) {
 }
 
 bool CheckedRegionFinder::VisitCallExpr(CallExpr *C) {
+  // TODO this should really be moved to the other visitor
+  // both for clarity and because it will allow it take advantage of
+  // redundancy checks
   auto FD = C->getDirectCallee();
+  FoldingSetNodeID ID;
+  C->Profile(ID, *Context, true);
   if (FD && FD->isVariadic()) {
-    // If this variadic call is in statement positon, we can wrap in it
-    // an unsafe block and avoid polluting the entire block as unsafe.
-    // If it's not (as in it is used in an expression) then we fall back to
-    // reporting an WILD value.
-    if (isInStatementPosition(C)) {
-      // Insert an _Unchecked block around the call
-      auto Begin = C->getBeginLoc();
-      Writer.InsertTextBefore(Begin, "_Unchecked { ");
-      auto End = C->getEndLoc();
-      Writer.InsertTextAfterToken(End, "; }");
-    } else {
-      // Call is inside an epxression, mark WILD.
-      Nwild++;
+    Wild = true;
+  } else {
+    if (FD) {
+      auto type = FD->getReturnType();
+      Wild |= (!(FD->hasPrototype() || FD->doesThisDeclarationHaveABody()))
+        || containsUncheckedPtr(type)
+        || (any_of(FD->param_begin(), FD->param_end(), [this] (Decl *param) {
+              auto var = Info.getVariable(param, Context);
+              return isWild(var);
+            }));
     }
+    handleChildren(C->children());
```

Comment:
As a practical matter, the children are expressions, so the only statements you'd find are statement expressions? As such, is this really needed, since they will all be wild, which is the default?

Reply:
It's needed because the sub expressions could use an unchecked value (such as a global) declared outside of the given scope.